development/build/utils.js

```diff
@@ -70,10 +70,42 @@ function getBrowserVersionMap(platforms, version) {
     const versionParts = [major, minor, patch];
     const browserSpecificVersion = {};
     if (prerelease) {
-      if (platform === 'firefox') {
-        versionParts[2] = `${versionParts[2]}${buildType}${buildVersion}`;
-      } else {
-        versionParts.push(buildVersion);
+      const { id } = loadBuildTypesConfig().buildTypes[buildType];
+      if (id < 10 || id > 64 || buildVersion < 0 || buildVersion > 999) {
+        throw new Error(
+          `Build id must be 10-64 and release version must be 0-999
+(inclusive). Received an id of '${id}' and a release version of
+'${buildVersion}'.
+
+Wait, but that seems so arbitrary?
+==================================
+
+We encode the build id and the release version into the extension version by
+concatenating the two numbers together. The maximum value for the concatenated
+number is 65535 (a Chromium limitation). The value cannot start with a '0'. We
+utilize 2 digits for the build id and 3 for the release version. This affords us
+55 release types and 1000 releases per 'version' + build type (for a minimum
```

Comment:
I think this math (50 and 1000) is right. Someone verify please!

Reply:
Sounds about right to me!